packages/webAuthn/background/webAuthn.ts

```diff
@@ -0,0 +1,226 @@
+import {
+  runtime,
+  tabs,
+  windows,
+  type Runtime,
+  type Tabs,
+  type Windows
+} from "webextension-polyfill"
+
+import json from "~packages/util/json"
+import { PortName } from "~packages/webAuthn/tabs/port"
+import {
+  isWebAuthnError,
+  type WebAuthnAuthentication,
+  type WebAuthnCreation,
+  type WebAuthnError,
+  type WebAuthnParams,
+  type WebAuthnRegistration,
+  type WebAuthnRequest
+} from "~packages/webAuthn/typing"
+
+export const webAuthnCreationAsyncWindow = async (
+  webAuthnCreation?: WebAuthnCreation
+): Promise<WebAuthnRegistration> => {
+  const createWindowUrl = `${runtime.getURL(
+    "tabs/createWebAuthn.html"
+  )}?user=${encodeURI(
+    webAuthnCreation?.user
+  )}&challengeCreation=${webAuthnCreation?.challenge}`
+
+  return (await webAuthnWindowOrTabAsync(
+    createWindowUrl,
+    true
+  )) as WebAuthnRegistration
+}
+
+export const webAuthnRequestAsyncWindow = async (
+  webAuthnRequest: WebAuthnRequest
+): Promise<WebAuthnAuthentication> => {
+  const requestWebauthnUrl = `${runtime.getURL(
+    "tabs/requestWebauthn.html"
+  )}?credentialId=${webAuthnRequest.credentialId}&challengeRequest=${
+    webAuthnRequest.challenge
+  }`
+
+  return (await webAuthnWindowOrTabAsync(
+    requestWebauthnUrl,
+    true
+  )) as WebAuthnAuthentication
+}
+
+export const webAuthnAsyncWindow = async (
+  tabId: number,
+  { webAuthnCreation, webAuthnRequest }: WebAuthnParams
+): Promise<WebAuthnRegistration | WebAuthnAuthentication> => {
+  const webAuthnUrl = `${runtime.getURL(
+    "tabs/webAuthn.html"
+  )}?tabId=${tabId}&user=${encodeURI(
+    webAuthnCreation?.user
+  )}&challengeCreation=${webAuthnCreation?.challenge}&credentialId=${
+    webAuthnRequest.credentialId
+  }&challengeRequest=${webAuthnRequest.challenge}`
+
+  return await webAuthnWindowOrTabAsync(webAuthnUrl, true)
+}
+
+export const webAuthnCreationAsyncTab = async (
```

Comment:
nit: At this moment we don't need tab, but there are lots of functions related to tab, which makes the code a bit messy 😢